tensorboard/util/test_util.py

```diff
@@ -273,3 +275,40 @@ def ensure_tb_summary_proto(summary):
     return summary
 
   return summary_pb2.Summary.FromString(summary.SerializeToString())
+
+
+def _run_conditionally(guard, name, default_reason=None):
+  """Creates a decorator that skips a test when guard returns False.
+
+    Args:
+      guard: A lambda that returns True if a test should be executed.
+      name: A human readable name for the decorator for an error message.
+      default_reason: A string describing why a test should be skipped. If it
+          is None, the decorator will make sure the reason is supplied by the
+          consumer of the decorator. Default is None.
+
+    Raises:
+      ValueError when both reason and default_reason are None.
```

Comment:
No, this function never raises `ValueError`; it returns a _function_
that may raise `ValueError`. You can tell that this must be the case
because this function has no argument `reason`.

Reply:
Hmm what is the right way to capture that in pydoc? It sounds like there should be a separate pydoc for the `_impl`?

Reply:
Good question. I don’t have a definitive answer. It would be fine me to
just clearly document the semantics of the returned function in the
“Returns” clause of the top-level function’s docs (like: “Returns: A
function that takes these arguments, raises under these conditions, and
returns a decorator.”). You might also consider adding docs to `_impl`
itself, which will appear if a user runs `help(run_v1_only)` or
`run_v1_only.__doc__` in an interactive session.